client/src/App.jsx

```
@@ -0,0 +1,57 @@
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  Navigate,
+} from 'react-router-dom';
+import './App.css';
+import Login from './components/Login';
+import ChatRoom from './components/ChatRoom';
+import Rooms from './components/Rooms';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { useState } from 'react';
+import { useEffect } from 'react';
+import { createUser } from './api/users';
+
+const queryClient = new QueryClient();
+
+export default function App() {
+  const [username, setUsername] = useState(
+    () => localStorage.getItem('username') || '',
+  );
+
+  useEffect(() => {
+    const saveUser = async () => {
+      const trimmed = username.trim();
+      if (!trimmed) return;
+
+      localStorage.setItem('username', trimmed);
+
+      try {
+        await createUser(trimmed);
```

You `await createUser(trimmed)` inside a try/catch, but the API helper `createUser` currently swallows errors (logs them) instead of rethrowing or returning a failure result. That means this catch block will never run for server-side failures (e.g. 400 'User already exists'). Consider making `createUser` propagate errors (or return a status) so the app can handle duplicate-user or network errors and show a message to the user. See the users API implementation for reference.